pkg/reconciler/broker/broker.go

```diff
@@ -428,6 +435,55 @@ func (r *Reconciler) reconcileChannel(ctx context.Context, channelResourceInterf
 	return channelable, nil
 }
 
+func (r *Reconciler) reconcileBrokerChannelEventPolicies(ctx context.Context, b *eventingv1.Broker, triggerChan *duckv1.Channelable, featureFlags feature.Flags) error {
+	expected := resources.MakeEventPolicyForBackingChannel(b, triggerChan)
+
+	if featureFlags.IsOIDCAuthentication() {
+		// get the eventpolicy , create if not exists.
+		foundEP, err := r.eventPolicyLister.EventPolicies(expected.Namespace).Get(expected.Name)
+		if apierrs.IsNotFound(err) {
+			// create the EventPolicy if it doesn't exists.
+			logging.FromContext(ctx).Info("Creating EventPolicy for Broker %s", expected.Name)
+
+			_, err = r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Create(ctx, expected, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create EventPolicy for Broker %s: %w", expected.Name, err)
+			}
+		} else if err != nil {
+			return fmt.Errorf("failed to get EventPolicy for Broker %s: %w", expected.Name, err)
+		} else if r.policyNeedsUpdate(foundEP, expected) {
+			// update the EventPolicy if it exists and needs update.
+			logging.FromContext(ctx).Info("Updating EventPolicy for Broker %s", expected.Name)
+			expected.SetResourceVersion(foundEP.GetResourceVersion())
+			_, err = r.eventingClientSet.EventingV1alpha1().EventPolicies(expected.Namespace).Update(ctx, expected, metav1.UpdateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to update EventPolicy for Broker %s: %w", expected.Name, err)
+			}
+		}
+	} else {
```

Comment:
all of these if/else blocks can make it a little unclear what is going on, I might refactor this to return early in many of these cases. For example, if the EventPolicy is successfully created, we can return `nil`. This gets rid of the need for the `else if r.policyNeedsUpdate(...)`, which can be turned into a simpler `if r.policyNeedsUpdate(...)`.

Reply:
good point, thanks for the feedback, I will make the changes!

Reply:
done, PTAL!